runtime/tests/decimal_conv.c

```diff
@@ -51,6 +51,44 @@ void testDecToFloat() {
     validatDecToFloat("-1.7976931348623156e+309", -INFINITY);
 }
 
+void validateDecToIntNonOverflow(const char *decStr, int64_t val) {
+    DecToIntResult res = _bal_decimal_to_int(_bal_decimal_const(decStr));
+    assert(res.val == val);
+    assert(!res.overflow);
+}
+
+void validateDecToIntOverflow(const char *decStr) {
+    DecToIntResult res = _bal_decimal_to_int(_bal_decimal_const(decStr));
+    assert(res.overflow);
+}
+
+void testDecToInt() {
+    validateDecToIntNonOverflow("1", 1);
+    validateDecToIntNonOverflow("0", 0);
+    validateDecToIntNonOverflow("-1", -1);
+    validateDecToIntNonOverflow("0.5", 0);
+    validateDecToIntNonOverflow("1.5", 2);
+    validateDecToIntNonOverflow("2.5", 2);
+    validateDecToIntNonOverflow("-0.5", 0);
+    validateDecToIntNonOverflow("-1.5", -2);
+    validateDecToIntNonOverflow("-2.5", -2);
+    validateDecToIntNonOverflow("1e2", 100);
+    validateDecToIntNonOverflow("1.5e2", 150);
+    validateDecToIntNonOverflow("-1.5e2", -150);
+    validateDecToIntNonOverflow("1.51e2", 151);
+    validateDecToIntNonOverflow("1.513e2", 151);
+    validateDecToIntNonOverflow("1.515e2", 152);
+    validateDecToIntNonOverflow("9223372036854775807", 9223372036854775807);
```

Comment:
Should test 9223372036854775807.4 and similar values.

Reply:
Added more tests